packages/providers/src/provider.ts

```diff
@@ -326,33 +326,23 @@ export default class Provider {
   /**
    * Updates the URL for the provider and fetches the consensus parameters for the new URL, if needed.
    */
-  async switchUrl(url: string) {
+  async connect(url: string) {
     this.url = url;
     this.createOperations();
     await this.fetchChainAndNodeInfo();
   }
 
   /**
-   * Retrieves and caches chain and node information if not already cached.
-   *
-   * - Checks the cache for existing chain and node information based on the current URL.
-   * - If not found in cache, fetches the information, caches it, and then returns the data.
+   * Fetches both the chain and node information, saves it to the cache, and return it.
    *
    * @returns NodeInfo and Chain
    */
   async fetchChainAndNodeInfo() {
-    let nodeInfo = Provider.nodeInfoCache[this.url];
-    let chain = Provider.chainInfoCache[this.url];
-
-    if (!nodeInfo) {
-      nodeInfo = await this.fetchNode();
-      Provider.nodeInfoCache[this.url] = nodeInfo;
-    }
+    const chain = await this.fetchChain();
+    const nodeInfo = await this.fetchNode();
 
-    if (!chain) {
-      chain = await this.fetchChain();
-      Provider.chainInfoCache[this.url] = chain;
-    }
+    Provider.chainInfoCache[this.url] = chain;
```

Comment:
Rather than setting the cache here, it's more appropriate to set the relevant caches in the `fetchChain`/`fetchNode` methods. Currently as they are, when somebody calls `fetchChain` or `fetchNode` independently, they might get new data that differs from the cached data, but the cache won't be updated.

Reply:
I agree @nedsalk , can do one of this 2 things:

1- remove `fetchChain` and `fetchNode` (they are new and will break almost no one). do a single query to get both chain and node info, then populate both caches in this method `fetchChainAndNodeInfo`.

OR

2- remove `fetchChainAndNodeInfo` and keep each caching separated as @nedsalk suggested.

Reply:
There is a third option: make `fetchChain` and `fetchNode` cache the data, and have `fetchChainAndNodeInfo` just be a wrapper around calls to both methods. This wouldn't break anyone and would be quite easy to implement. I'll request changes for this part.